stdlib/private/StdlibCollectionUnittest/CheckCollectionType.swift

```diff
@@ -1254,6 +1260,149 @@ extension TestSuite {
       }
     }
 
+    //===------------------------------------------------------------------===//
+    // last(where:)
+    //===------------------------------------------------------------------===//
+
+    let lastPerformanceTest = FindTest(
+      expected: 3,
+      element: 2020,
+      sequence: [ 1010, 2020, 3030, 2020, 4040 ],
+      expectedLeftoverSequence: [ 3030, 2020, 4040 ])
+  
+    self.test("\(testNamePrefix).last(where:)/semantics") {
+      for test in findLastTests {
+        let c = makeWrappedCollectionWithEquatableElement(test.sequence)
+        let closureLifetimeTracker = LifetimeTracked(0)
+        let found = c.last(where: {
+          _blackHole(closureLifetimeTracker)
+          return $0 == wrapValueIntoEquatable(test.element)
+        })
+        expectEqual(
+          test.expected == nil ? nil : wrapValueIntoEquatable(test.element),
+          found,
+          stackTrace: SourceLocStack().with(test.loc))
+        if let expectedIdentity = test.expected {
+          expectEqual(
+            expectedIdentity, extractValueFromEquatable(found!).identity,
+            "last(where:) should find only the first element matching its predicate")
+        }
+      }
+    }
+
+    self.test("\(testNamePrefix).last(where:)/performance") {
+      let test = lastPerformanceTest
+      let closureLifetimeTracker = LifetimeTracked(0)
+      expectEqual(1, LifetimeTracked.instances)
+      let c = makeCollectionOfEquatable(test.sequence.map(wrapValueIntoEquatable))
+      var closureCounter = 0
+      let found = c.last(where: {
+        (candidate) in
+        _blackHole(closureLifetimeTracker)
+        closureCounter += 1
+        return
+          extractValueFromEquatable(candidate).value == test.element.value
+      })
+      expectEqual(
+        test.expected == nil ? nil : wrapValueIntoEquatable(test.element),
+        found,
+        stackTrace: SourceLocStack().with(test.loc))
+      expectEqual(
+        2,
+        closureCounter,
+        stackTrace: SourceLocStack().with(test.loc))
+    }
+
+    //===------------------------------------------------------------------===//
+    // lastIndex(of:)/lastIndex(where:)
+    //===------------------------------------------------------------------===//
+
+    self.test("\(testNamePrefix).lastIndex(of:)/semantics") {
+      for test in findLastTests {
+        let c = makeWrappedCollectionWithEquatableElement(test.sequence)
+        var result = c.lastIndex(of: wrapValueIntoEquatable(test.element))
+        expectType(
+          Optional<CollectionWithEquatableElement.Index>.self,
+          &result)
+        let zeroBasedIndex = result.map {
+          numericCast(c.distance(from: c.startIndex, to: $0)) as Int
```

Comment:
This is just a long version of saying `Int(c.distance(from: c.startIndex, to: $0))`, isn't it?

Reply:
... and it should be `Int` anyway, since the deprecation of `Collection.IndexDistance`.

Reply:
This is pervasive in these files, from when `IndexDistance` was a thing. I'll fix 'em in a follow-up.